Assets/ModScripts/Submodules/SimonSkips.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Security.Principal;
+using UnityEngine;
+using KModkit;
+using Random = UnityEngine.Random;
+
+
+
+public class SimonSkips : Puzzle
+{
+    int[] arrowColours;
+    int[] orderedArrows;
+    List<int> finalSequence = new List<int>(); // List of indexes to press
+    List<int> inputtedSequence = new List<int>();
+    bool submitEmpty = false;
+    int inputNum = 0;
+
+    public SimonSkips(CruelModkitScript Module, int ModuleID, ComponentInfo Info, bool Vanilla, byte Components) : base(Module, ModuleID, Info, Vanilla, Components)
+    {
+        Debug.LogFormat("[The Cruel Modkit #{0}] Solving Simon Skips. Press the ❖ button to initiate the module.", ModuleID);
+        int[] newArrows = new int[] { 4, 6, 7, 8, 9 }.Shuffle().Take(4).ToArray();
+
+        for (int i = 4; i < 8; i++)
+        {
+            Info.Arrows[i] = newArrows[i-4];
+        }
+        for (int i = 4; i < 8; i++)
+        {
+            Module.Arrows[i].GetComponentInChildren<Renderer>().material = Module.ArrowMats[Info.Arrows[i]];
+            Module.Arrows[i].transform.Find("ArrowLight").GetComponentInChildren<Light>().color = Info.ArrowLightColors[Info.Arrows[i]];
+            Module.Arrows[i].transform.Find("ArrowLight").GetComponentInChildren<Light>().intensity += (Info.Arrows[i] == 8) ? 10 : 0;
+        }
+
+        orderedArrows = new int[] { Info.Arrows[0],  Info.Arrows[4], Info.Arrows[1], Info.Arrows[5], Info.Arrows[2], Info.Arrows[6], Info.Arrows[3], Info.Arrows[7] };
+        finalSequence.Add(FindStartingColor());
+        if (finalSequence[0] != 8)
+            Debug.LogFormat("[The Cruel Modkit #{0}] The starting colour is {1}.", ModuleID, ComponentInfo.ArrowColors[orderedArrows[finalSequence[0]]]);
+        else
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] The starting colour is white or black. Submit an empty sequence!", ModuleID);
+            submitEmpty = true;
+        }
+    }
+
+    int FindStartingColor()
+    {
+        int product = Module.Bomb.GetSerialNumberNumbers().Where(x => x != 0).Aggregate(1, (a, b) => a * b);
+        product += Info.NumberDisplay;
+        product %= 8;
+        if (orderedArrows[product] > 7)
+            return 8;
+        return product;
+    }
+
+    void FindFullSequence()
+    {
+        for (int i = 1; i < 9; i++)
```

Comment:
You only use i-1. Why not initialize i by 0 instead of 1?
```suggestion
        for (int i = 0; i < 8; i++)
```